internal/app/connectconformance/server_runner.go

```diff
@@ -175,8 +175,8 @@ func runTestCasesForServer(
 			default:
 				results.assert(name, expectations[resp.TestName], resp.GetResponse())
 			}
-			if isReferenceClient {
-				for _, msg := range resp.GetFeedback() {
+			if isReferenceClient && resp != nil {
+				for _, msg := range resp.Feedback {
```

Comment:
Unrelated tweak: we end up using `resp.TestName` below (without accessor), so it seemed more obviously correct to hoist the nil check up. And then we can safely use direct field access in this block.